### PR TITLE
feat(core): plugin_key_from_action_key helper (ADR-0095 D1 U-D1.1)

### PR DESCRIPTION
**Pull request by vanyastaff**
## What

First unit of the **ADR-0095 D1** decomposition (the units that make the U3 trigger-dispatch slice live in production). Adds a pure helper to `nebula-core`:

```rust
pub fn plugin_key_from_action_key(action: &ActionKey) -> Result<PluginKey, KeyParseError>
```

It extracts the plugin namespace from an action key: the segment **before the first `.`** of `{plugin_key}.{action_name}`; a key with no `.` is itself the plugin-key candidate; the `PluginKey` validation error (incl. empty prefix → `Err`) propagates from `PluginKey::new`.

This mirrors the namespacing invariant already enforced at plugin registration (`ResolvedPlugin`'s `{plugin}.` prefix + `starts_with` check) — the *validation* stays in `nebula-plugin`; this adds the *extraction*, in `nebula-core` where both `ActionKey` and `PluginKey` are defined (one fact, one place).

## Why a free fn (not `PluginKey::from_action_key`)

`PluginKey` is a type alias over the external `domain_key::Key<PluginDomain>`, so an inherent `impl` is forbidden by the orphan rule (E0116). A free fn re-exported via the crate prelude is the idiomatic shape.

## Where it fits

D1 decomposition: **U-D1.1** (this) → U-D1.2 (registry-backed `RoutingResolver` that consumes this to map a workflow/trigger → `required_plugin_key`) → U-D1.3 (`nebula-worker`) → … It is **tested-but-uncalled by design** until U-D1.2 lands (same pattern as D6's resolver). Independent of the in-flight atomic-fold (U0) — different crate, different files.

## Evidence

- `cargo nextest run -p nebula-core`: 64/64 pass (4 new tests: namespaced extraction, first-dot-only split, no-dot whole-string, empty-prefix → `Err`).
- `cargo clippy -p nebula-core --all-targets --all-features -- -D warnings`: clean.
- `RUSTDOCFLAGS=-D warnings cargo doc --no-deps -p nebula-core`: clean.
- No `unwrap`/`expect`/`panic!` in library code.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Updated internal architectural decision records and memory catalogs documenting plugin system patterns, trigger dispatch mechanisms, dependency resolution strategies, and resource lifecycle management.

* **New Features**
  * Added a utility helper function for parsing and extracting plugin identifiers from action key formats.

* **Tests**
  * Extended test suite with comprehensive coverage for key parsing operations and edge case validation.

### CODE REVIEW

File: crates/core/src/keys.rs [hunk 1]
```diff
@@ -43,6 +43,35 @@ key_type!(pub PluginKey, PluginDomain);
...
+/// The candidate is validated through the same constructor `PluginKey::new` uses,
+/// so all plugin-key invariants (character set, length, structure) are enforced.
+///
+/// # Errors
+///
+/// Returns [`KeyParseError`] when the extracted prefix is not a valid plugin key
+/// (e.g. empty, contains illegal characters, or exceeds the maximum length).
+pub fn plugin_key_from_action_key(action: &ActionKey) -> Result<PluginKey, KeyParseError> {
+    let prefix = match action.as_str().split_once('.') {
+        Some((before, _)) => before,
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply crates/core/src/keys.rs:69]: Valid — confirmed. `PluginKey` uses the default `domain_key` charset, which permits `.` (the same charset as `ActionKey`, which must allow it for `{plugin}.{action}`). So a legal dotted plugin key (e.g. `plugin.group` with action `plugin.group.action`) would have this first-`.` split return `plugin` — the wrong `required_plugin_key` → an unclaimable job. No dotted plugin keys exist in the tree today (grep is empty), so it's latent, but the first-`.` split is unsound as a general routing primitive.

Resolving via one of:
(a) **forbid `.` in `PluginKey`'s domain** — makes the `{plugin}.{component}` separator unambiguous and the first-`.` split provably correct (structural / type-enforced; aligns with the curated single-token plugin-key model), or
(b) **registry longest-matching-prefix** in the D1 routing resolver — correct regardless of dots, but moves the extraction out of this pure helper.

Leaning (a). The dot-policy is an owner decision (whether hierarchical/dotted plugin keys are ever intended); pinning it before this helper is used as a routing key. Not shipping the first-`.` split as-is until then. Good catch.

[the remark this answers is not in the document]
Comment (vanyastaff) [reply crates/core/src/keys.rs:69]: Resolution (owner clarified): an `ActionKey` does not store a composite `{plugin}.{action}` key — the action has its own identity; the plugin key is not embedded in it. So deriving the plugin key by string-splitting the action key is the wrong approach entirely (not just the dot edge-case). The D1 routing resolver will instead resolve the owning plugin via `PluginRegistry` lookup (authoritative, dot-agnostic). This helper PR is being closed; no `PluginKey` dot-policy change is needed. Thanks for catching it — it surfaced the deeper modeling error.

**Comment by vanyastaff** [comment]
Closing — superseded by an owner design clarification.

The premise of this helper (derive the plugin key by string-splitting an action key at the first `.`) is wrong: an `ActionKey` is the action's own identity and does **not** store a composite `{plugin}.{action}` key — there is no plugin key embedded in it to extract. The plugin↔action relationship is a **registry fact** (which plugin registered which action), not a property of the action-key string.

So codex's P2 (dotted plugin keys break the first-`.` split) is moot the right way: we don't parse the action key at all. The D1 routing resolver (U-D1.2, `nebula-engine`) will resolve the owning plugin via `PluginRegistry` lookup, which is authoritative and dot-agnostic — no `PluginKey` dot-policy change needed.

The `.gitignore` hygiene fix from this PR (untrack `.claude/agent-memory/`) will be re-landed separately; `main` was never polluted (those files only ever lived on this branch).
